wcs-functions.php

```diff
@@ -149,62 +152,32 @@ function wcs_create_subscription( $args = array() ) {
 		return new WP_Error( 'woocommerce_subscription_invalid_billing_interval', __( 'Invalid subscription billing interval given. Must be an integer greater than 0.', 'woocommerce-subscriptions' ) );
 	}
 
-	$subscription_data['post_type']     = 'shop_subscription';
-	$subscription_data['post_status']   = 'wc-' . apply_filters( 'woocommerce_default_subscription_status', 'pending' );
-	$subscription_data['ping_status']   = 'closed';
-	$subscription_data['post_author']   = 1;
-	$subscription_data['post_password'] = uniqid( 'order_' );
-	// translators: Order date parsed by strftime
-	$post_title_date = strftime( _x( '%b %d, %Y @ %I:%M %p', 'Used in subscription post title. "Subscription renewal order - <this>"', 'woocommerce-subscriptions' ) ); // phpcs:ignore WordPress.WP.I18n.UnorderedPlaceholdersText
-	// translators: placeholder is order date parsed by strftime
-	$subscription_data['post_title']    = sprintf( _x( 'Subscription &ndash; %s', 'The post title for the new subscription', 'woocommerce-subscriptions' ), $post_title_date );
-	$subscription_data['post_date_gmt'] = $args['date_created'];
-	$subscription_data['post_date']     = get_date_from_gmt( $args['date_created'] );
+	$subscription = new \WC_Subscription();
+	$subscription->set_status( $args['status'] );
```

Comment:
It's possible that the status isn't actually passed in and so this could be `''`. We should do something similar to how `wc_create_order` handle the setting by only calling the setters if it's set and it validates.

Reply:
I've wrapped `set_status()` in an `if`